memory/table.go

```diff
@@ -532,6 +537,30 @@ func (t *Table) addColumnToSchema(ctx *sql.Context, newCol *sql.Column, order *s
 		newSchCol.Default = newDefault.(*sql.ColumnDefaultValue)
 	}
 
+	if newCol.AutoIncrement {
+		t.autoColIdx = newColIdx
+		t.autoIncVal = 0
+
+		if newColIdx < len(t.schema.Schema) {
```

Comment:
after we drop but before we add, does the schema length change? if not this is fine, otherwise maybe one off error

Reply:
You can't add an auto increment column, only modify an existing one, so this should be fine

Reply:
modify column calls drop and add in succession, and i'd think the schema is smaller between the two
```go
	_ = t.dropColumnFromSchema(ctx, columnName)
	t.addColumnToSchema(ctx, column, order)
```